### DISCUSSION TITLE
starlette >v0.18 requires status_code on Response init

### DISCUSSION THREAD

**Discussion by syldman** [fastapi/fastapi#6375] [Questions] [answered]
### Example Code

```python
https://github.com/tiangolo/fastapi/blob/master/fastapi/dependencies/utils.py

line 465
    response = response or Response(
        content=None,
        status_code=None,  # type: ignore
        headers=None,  # type: ignore # in Starlette
        media_type=None,  # type: ignore # in Starlette
        background=None,  # type: ignore # in Starlette
    )
```

the above fails as https://github.com/encode/starlette/blob/master/starlette/responses.py requires status_code to be INT

```
line 74
        if (
            body is not None
            and populate_content_length
            and not (self.status_code < 200 or self.status_code in (204, 304))
        ):
```

### FastAPI Version
0.75.1

**Comment by jgould22** [comment]
Yes, this is one of the reasons FastAPI's current code base uses Starlette 17.

This fix is applied for the update merge request.
https://github.com/tiangolo/fastapi/pull/4483/files#diff-aef3dac481b68359f4edd6974fa3a047cfde595254a4567a560cebc9ccb0673f

**Comment by deJagerF** [comment]
Greetings

Can I suggest that you state on the the home page that you use Starlette 17 and that users must be carefull when doing an upgrade to their python packages

I upgraged my environment, then everything broke till I reinstalled fastapi[all] and I noticed that Starlette is being downgraded to 17

**Comment by jgould22** [comment]
The projects dependencies are listed in the [pyproject.toml](https://github.com/tiangolo/fastapi/blob/master/pyproject.toml#L38).

Are you not using a vituralenv ? or perhaps listing Starlette along with FastAPI as a dependency?

**Comment by deJagerF** [comment]
Greetings

I used pip to install FastAPI, (with all optional dependencies) then later
I did an update to all packages.  That is when I noticed that my
application was failing.  I then did a reinstall of FastAPI and that is
when I noticed the downgrade. I checked the home page but did not see the
requirement, it was only when I searched git that I saw the post and the
reply with the requirement

That is why I made the suggestion that it be listed on the homepage

Hope this helps

Regards

On Fri, 22 Apr 2022, 15:56 Jordan Gould, ***@***.***> wrote:

> The projects dependencies are listed in the pyproject.toml.
>
> Are you not using a vituralenv ? or perhaps listing Starlette along with
> FastAPI as a dependency?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/tiangolo/fastapi/issues/4800#issuecomment-1106542501>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHYI5DM27RPDW4DL7POQMW3VGKVYFANCNFSM5TRCURZQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

-- 
This message and attachments are subject to a disclaimer.

Please refer to 
http://upnet.up.ac.za/services/it/documentation/docs/004167.pdf 
<http://upnet.up.ac.za/services/it/documentation/docs/004167.pdf> for
full 
details.

**Comment by Kludex** [comment, marked as answer]
- https://github.com/tiangolo/fastapi/pull/4483

**Comment by raphaelauv** [comment]
@syldman can you close the issue -> https://github.com/tiangolo/fastapi/releases/tag/0.76.0
